### PR TITLE
Add doc starter

### PR DESCRIPTION
**Pull request by codetent**
## What's Changed

This pull request adds a documentation starter published to GHA.
Additionally, it introduces release-drafter for the release process and version calculation.

## Todo

- [x] The pull request title is in the following format: `Add new feature`
- [x] A label (`breaking`, `feature`, `bug`, `chore`) indicating the type of the change has been added

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+For updating mocks, execute:
+
+```
+mockery
+```
+
+## Documentation
+
+The documentation of the project is hosted using GitHub pages.
+For that, a static website is built based one Markdown files located in `docs/pages`.
```
Comment (maruu) [inline CONTRIBUTING.md:104]:
```suggestion
For that, a static website is built based on Markdown files located in `docs/pages`.
```

File: docs/pages/getting-started/installation.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+```bash
+curl -L -o crypta "https://github.com/codetent/crypta/releases/latest/download/crypta_Darwin_x86_64" && sudo install -c -m 0755 crypta /usr/local/bin
+```
+
+</TabItem>
+<TabItem value="linux">
+
+```bash
+curl -L -o crypta "https://github.com/codetent/crypta/releases/latest/download/crypta_Linux_x86_64" && sudo install -c -m 0755 crypta /usr/local/bin
+```
```
Comment (maruu) [inline docs/pages/getting-started/installation.mdx:35]: I think the uploaded artifacts currently only contain archives, and not the executables themselves. How does this work?

**Review by maruu — APPROVED** [review]
nice! just some minor remarks

File: docs/pages/getting-started/installation.mdx (hunk 2 above)
Comment (codetent) [reply docs/pages/getting-started/installation.mdx]: I skipped archiving for now and uploaded the binaries directly. Additionally, I updated the documenation

File: docs/hack/cli/main.go [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
 
 import (
+	"path/filepath"
+	"strings"
+
 	"github.com/codetent/crypta/cmd"
 	"github.com/spf13/cobra/doc"
 )
 
+func filePrepender(s string) string {
```
[the remark this answers is not in the document]
Comment (codetent) [reply docs/hack/cli/main.go:11]: Using a generic function name makes probably sense when we add more functionality but for now, you are right, changing the function name makes sense.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+### Merge Verification
+
+For each pull request that is going to be merged to the `main` branch, a merge verification is automatically executed.
+The following steps are executed:
+
+1. Run statical analysis
+2. Execute unit tests
+3. Build binaries as snapshots
+4. Build documentation
```
[the remark this answers is not in the document]
Comment (codetent) [reply CONTRIBUTING.md:123]: I removed the workflow description at the build section and only mentioned the necessary step.

**Pull request merged**